pandas/tests/indexes/test_base.py

```diff
@@ -305,6 +305,15 @@ def test_constructor_simple_new(self):
         result = idx._simple_new(idx, 'obj')
         tm.assert_index_equal(result, idx)
 
+    def test_constructor_names(self):
+        idx = Index([1, 2, 3], name='a')
+        assert idx.name == 'a'
+        assert idx.names == ('a',)
```

Comment:
need testing for each index type & multiindex. need to hit each of those error conditions (e.g. name and/or names not None). we *may* already have some of these tests but should consolidate.